PhpcrMigration/Infrastructure/Repository/EntityRepository.php

```diff
@@ -193,4 +210,53 @@ public function addOrUpdateChildNode(array $data, string $tableName, array $type
 
         $this->connection->insert($tableName, $data, $types);
     }
+
+    private function getNextIdValue(string $tableName): ?int
+    {
+        $result = null;
+
+        $platform = $this->connection->getDatabasePlatform();
+        $sequences = $this->getSequences();
+        foreach ($sequences as $sequence) {
+            $sequenceName = $sequence->getName();
+            if (\str_contains($sequenceName, $tableName)) {
+                /** @var int|null|false $result */
+                $result = $this->connection->fetchOne(
+                    $platform->getSequenceNextValSQL($sequenceName)
+                );
+
+                if (false === $result || null === $result) {
+                    throw new \RuntimeException('Failed to get next ID value from sequence' . $sequenceName);
+                }
+
+                break;
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * @return Sequence[]
+     */
+    private function getSequences(): array
+    {
+        if (null !== $this->sequences) {
+            return $this->sequences;
+        }
+
+        $sequences = [];
+        try {
+            $sequences = $this->connection->createSchemaManager()->listSequences();
+        } catch (Exception) {
+            // @ignoreException
+        }
+
+        return $this->sequences = $sequences;
+    }
+
+    public function reset(): void
+    {
+        $this->sequences = null;
+    }
```

Comment:
```suggestion
    public function reset(): void
    {
        $this->sequences = null;
    }
```

requires a `kernel.reset` service tag ;)